mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -842,9 +852,25 @@ class LogisticRegression @Since("1.2.0") (
             denseCoefficientMatrix.update(classIndex, featureIndex,
               value / featuresStd(featureIndex))
           }
+          // It seems that we only adjust coef, intercept is missing?
           if (isIntercept) interceptVec.toArray(classIndex) = value
         }
 
+        // adjust intercept
+        if ($(fitIntercept)) {
+          var i = 0
+          while (i < denseCoefficientMatrix.numRows) {
+            var j = 0
+            var adjust = 0.0
+            while (j < numFeatures) {
+              adjust += denseCoefficientMatrix(i, j) * featuresMean(j)
```

Comment:
While I understand perfectly well what you are doing here, someone who hasnt thought about this will not.  I would add a comment explaining why the adjustment is equal to the dot product of the coefficients and the means.  Something along the lines of "If one applies a linear transformation to the data x -> (x - avg(x))/std(x), and finds a minimum of the objective function with the transformed data, then moving all the averages over to the intercept changes the intercept by coeffs(in unscaled space) dot averages."